public/components/custom_panels/forms/add_new_visualizations.tsx

```diff
@@ -0,0 +1,275 @@
+    );
+  };
+
+  useEffect(() => {
+    const previewTemplate =
+      isPreviewError == '' ? (
+        <>
+          <EuiFlexGroup>
+            <EuiFlexItem style={{ minHeight: '200' }}>
+              <PlotSample data={previewData} layout={{ showlegend: false }} />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </>
+      ) : (
+        <>
+          <div style={{ minHeight: '200', overflow: 'scroll', textAlign: 'center' }}>
+            <EuiSpacer size="l" />
+            <EuiIcon type="alert" color="danger" size="l" />
+            <EuiSpacer size="l" />
+            <EuiText>
+              <h2>Error in rendering the visualizaiton</h2>
+            </EuiText>
+            <EuiSpacer size="l" />
+            <EuiText>
+              <p>{isPreviewError}</p>
+            </EuiText>
+          </div>
+        </>
+      );
+
+    setPreviewArea(previewTemplate);
+  }, [previewLoading]);
+
+  const onChangeText = (
+    e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>,
+    setter: React.Dispatch<React.SetStateAction<string>>
+  ) => {
+    setter(e.target.value);
+  };
+
+  const advancedVisualization = () => {
+    //NOTE: Add Redux functions to pass pplquery and time filters to events page
+    window.location.assign('#/explorer/events');
+  };
+
+  const addVisualization = () => {
+    if (!isNameValid(newVisualizationTitle)) {
+      setToast('Invalid Visualization Name', 'danger');
+      return;
+    }
+
+    if (pplQuery.length == 0) {
+      setToast('Invalid PPL Query', 'danger');
+      return;
+    }
+
+    const newDimensions = getNewVizDimensions(panelVisualizations);
+    setPanelVisualizations([
+      ...panelVisualizations,
+      {
+        id: htmlIdGenerator()(),
+        title: newVisualizationTitle,
+        query: _.escape(pplQuery),
+        type: newVisualizationType,
+        ...newDimensions,
+      },
+    ]);
+
+    //NOTE: Add a backend call to add a visualization
+    setToast(`Visualization ${newVisualizationTitle} successfully added!`, 'success');
+    closeVizWindow();
+  };
+
+  return (
+    <>
+      <EuiSpacer size="l" />
+      <EuiFormRow
+        label="Visualization name"
+        helpText="Enter a unique and descriptive name between 1-50 characters."
+      >
+        <EuiFieldText
+          name="Name"
+          value={newVisualizationTitle}
+          onChange={(e) => onChangeText(e, setNewVisualizationTitle)}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        label="PPL Query"
+        helpText={
+          <Fragment>
+            Use [example commands] to draw visaulizations.{' '}
+            <EuiLink href="https://opensearch.org/docs/search-plugins/ppl/index/" target="_blank">
+              Learn More
+            </EuiLink>{' '}
+          </Fragment>
+        }
+        fullWidth={true}
+      >
+        <EuiTextArea
+          placeholder="Placeholder text"
+          aria-label="Use aria labels when no actual label is in use"
```

Comment:
should these be something like "PPL query"?

Reply:
I missed this. Thanks